src/osd/scrubber/scrub_resources.h

```diff
@@ -10,6 +10,36 @@
 #include "common/Formatter.h"
 #include "osd/osd_types.h"
 
+/*
+ * AsyncReserver for scrub 'remote' reservations
+ * -----------------------------------------------
+ *
+ * On the replica side, all reservations are treated as having the same priority.
+ * Note that 'high priority' scrubs, e.g. user-initiated scrubs, are not required
+ * to perform any reservations, and are never handled by the replicas' OSD.
+ *
+ * A queued scrub reservation request is cancelled by any of the following events:
+ *
+ * - a new interval: in this case, we do not expect to see a cancellation request
+ *   from the primary, so we can simply remove the request from the queue;
+ *
+ * - a cancellation request from the primary: probably a result of timing out on
+ *   the reservation process. Here, we can simply remove the request from the queue.
+ *
+ * - a new reservation request for the same PG: which means we had missed the
+ *   previous cancellation request. Possible reactions: asserting, or simply
+ *   removing the previous request from the queue;
```

Comment:
We should cancel the previous one and start the new one.  It should also log so that teuthology can fail the test since this should be impossible.